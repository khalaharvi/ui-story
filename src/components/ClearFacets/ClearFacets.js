import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { connectCurrentRefinements } from 'react-instantsearch/connectors';

import { createClassNames } from 'utils';

const cx = createClassNames('clear-facets');

export const ClearFacets = ({
  className,
  items,
  canRefine,
  refine
}) => (
  <div className={classnames(cx(''), className)}>
    <button
      className={cx('button', !canRefine && 'button--disabled')}
      onClick={() => refine(items)}
      disabled={!canRefine}
    >
      Clear All
    </button>
  </div>
);

ClearFacets.propTypes = {
  className: PropTypes.string,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  canRefine: PropTypes.bool.isRequired,
  refine: PropTypes.func.isRequired
};

export default connectCurrentRefinements(ClearFacets);
