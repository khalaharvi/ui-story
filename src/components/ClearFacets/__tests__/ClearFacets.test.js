import React from 'react';
import { mount, shallow } from 'enzyme';

import { ClearFacets } from '../ClearFacets';

describe('<ClearFacets />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ClearFacets
        items={[{ filter: 1 }]}
        refine={() => null}
        canRefine
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with className', () => {
    const wrapper = shallow(
      <ClearFacets
        items={[{ filter: 1 }]}
        refine={() => null}
        canRefine
        className="custom-clear-facets"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has a disabled mode', () => {
    const wrapper = shallow(
      <ClearFacets
        items={[]}
        refine={() => null}
        canRefine={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a disabled mode', () => {
    const wrapper = shallow(
      <ClearFacets
        items={[]}
        refine={() => null}
        canRefine={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should be disabled when there is no filters', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <ClearFacets
        items={[]}
        refine={refine}
        canRefine={false}
      />
    );

    expect(refine.mock.calls).toHaveLength(0);
    wrapper.find('button').simulate('click');
    expect(refine.mock.calls).toHaveLength(0);
  });

  it('should not be disabled when there are filters', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <ClearFacets
        items={[{ value: 'test', label: 'test: test' }]}
        refine={refine}
        canRefine
      />
    );

    expect(refine.mock.calls).toHaveLength(0);
    wrapper.find('button').simulate('click');
    expect(refine.mock.calls).toHaveLength(1);
  });
});
