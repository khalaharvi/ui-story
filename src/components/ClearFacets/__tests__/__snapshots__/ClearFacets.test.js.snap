// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearFacets /> has a disabled mode 1`] = `
<div
  className="dh-clear-facets"
>
  <button
    className="dh-clear-facets-button dh-clear-facets-button--disabled"
    disabled={true}
    onClick={[Function]}
  >
    Clear All
  </button>
</div>
`;

exports[`<ClearFacets /> renders correctly 1`] = `
<div
  className="dh-clear-facets"
>
  <button
    className="dh-clear-facets-button"
    disabled={false}
    onClick={[Function]}
  >
    Clear All
  </button>
</div>
`;

exports[`<ClearFacets /> renders correctly with className 1`] = `
<div
  className="dh-clear-facets custom-clear-facets"
>
  <button
    className="dh-clear-facets-button"
    disabled={false}
    onClick={[Function]}
  >
    Clear All
  </button>
</div>
`;

exports[`<ClearFacets /> should have a disabled mode 1`] = `
<div
  className="dh-clear-facets"
>
  <button
    className="dh-clear-facets-button dh-clear-facets-button--disabled"
    disabled={true}
    onClick={[Function]}
  >
    Clear All
  </button>
</div>
`;
